optimum/habana/diffusers/pipelines/stable_diffusion_xl/pipeline_stable_diffusion_xl.py

```diff
@@ -34,15 +38,629 @@
+    Sample Steps are Flawed](https://arxiv.org/pdf/2305.08891.pdf). See Section 3.4
+    """
+    std_text = noise_pred_text.std(dim=list(range(1, noise_pred_text.ndim)), keepdim=True)
+    std_cfg = noise_cfg.std(dim=list(range(1, noise_cfg.ndim)), keepdim=True)
+    # rescale the results from guidance (fixes overexposure)
+    noise_pred_rescaled = noise_cfg * (std_text / std_cfg)
+    # mix with the original results from guidance by factor guidance_rescale to avoid "plain looking" images
+    noise_cfg = guidance_rescale * noise_pred_rescaled + (1 - guidance_rescale) * noise_cfg
+    return noise_cfg
+
+# Copied from diffusers.models.unet_2d_condition.UNet2DConditionModel.set_attn_processor
+def set_attn_processor_hpu(self, processor: Union[AttentionProcessor, Dict[str, AttentionProcessor]]):
+    r"""
+    Sets the attention processor to use to compute attention.
+    Parameters:
+        processor (`dict` of `AttentionProcessor` or only `AttentionProcessor`):
+            The instantiated processor class or a dictionary of processor classes that will be set as the processor
+            for **all** `Attention` layers.
+
+            If `processor` is a dict, the key needs to define the path to the corresponding cross attention
+            processor. This is strongly recommended when setting trainable attention processors.
 
+    """
+    count = len(self.attn_processors.keys())
 
-logger = logging.get_logger(__name__)  # pylint: disable=invalid-name
+    if isinstance(processor, dict) and len(processor) != count:
+        raise ValueError(
+            f"A dict of processors was passed, but the number of processors {len(processor)} does not match the"
+            f" number of attention layers: {count}. Please make sure to pass {count} processor classes."
+        )
+
+    def fn_recursive_attn_processor(name: str, module: torch.nn.Module, processor):
+            if hasattr(module, "set_processor"):
+                if os.environ.get('PATCH_SDPA') is not None:
+                    setattr(module, 'attention_module', ScaledDotProductAttention())
+                    module.set_processor(processor(module.attention_module))
+                else:
+                    module.set_processor(processor.pop(f"{name}.processor"))
+
+            for sub_name, child in module.named_children():
+                fn_recursive_attn_processor(f"{name}.{sub_name}", child, processor)
+
+    for name, module in self.named_children():
+        fn_recursive_attn_processor(name, module, processor)
+
+# Copied from diffusers.models.unet_2d_condition.UNet2DConditionModel.set_default_attn_processor
+def set_default_attn_processor_hpu(self):
+    """
+    Disables custom attention processors and sets the default attention implementation from HPU.
+    """
+    processor = AttnProcessor2_0
+    set_attn_processor_hpu(self, processor)
+
+
+class StableDiffusionXLPipeline_HPU(StableDiffusionXLPipeline):
```

Comment:
StableDiffusionXLPipeline_HPU is not used in any example.

Reply:
StableDiffusionXLPipeline_HPU implementation is used by MLPERF SD XL workload published in model garden. So the usage in optimum-habana wont be visible. 

JH...!

Reply:
Let's move this new class and all the related changes in this file to a new file called `pipeline_stable_diffusion_xl_mlperf.py` please. It will be clearer and easier to maintain.